src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3844,33 +3848,36 @@ def main():
             jobutil = JobUtils(event.job)
             pbs.logmsg(pbs.EVENT_DEBUG4,
                        "%s: Job information class instantiated" %
-                       (caller_name()))
+                       caller_name())
         else:
             pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Event does not include a job" %
-                       (caller_name()))
+                       caller_name())
+        # Parse the cgroup configuration file here so we can use the file lock
+        cfg = CgroupUtils.parse_config_file()
         # Instantiate the cgroup utility class
         vnode = None
         if hasattr(event, 'vnode_list'):
             if hostname in event.vnode_list:
                 vnode = event.vnode_list[hostname]
-        cgroup = CgroupUtils(hostname, vnode)
-        pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Cgroup utility class instantiated" %
-                   (caller_name()))
-        # Bail out if there is nothing to do
-        if not cgroup.subsystems:
-            pbs.logmsg(pbs.EVENT_DEBUG,
-                       "%s: Cgroups disabled or none to manage" %
-                       (caller_name()))
-            event.accept()
-        # Call the appropriate handler
-        if hooks.invoke_handler(event, cgroup, jobutil) is True:
-            pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Hook handler returned success" %
-                       (caller_name()))
-            event.accept()
-        else:
-            pbs.logmsg(pbs.EVENT_DEBUG, "%s: Hook handler returned failure" %
-                       (caller_name()))
-            event.reject()
+            with Lock(cfg['cgroup_lock_file']):
```

Comment:
I think this block is indented too far. This portion of the code should be called regardless of whether the event has a vnode_list attribute.

Reply:
You are right. Fixed.